drivers/ethernet/eth_nxp_enet.c

```diff
@@ -0,0 +1,1050 @@
+	}
+
+	if (eir & ENET_EIR_MII_MASK) {
+		/* Callback to MDIO driver for relevant interrupt */
+		nxp_enet_mdio_callback(config->mdio, nxp_enet_interrupt);
+	}
+
+	irq_unlock(irq_lock_key);
+}
+
+
+/*
+ ******************
+ * Initialization *
+ ******************
+ */
+
+static int eth_nxp_enet_init(const struct device *dev)
+{
+	struct nxp_enet_mac_data *data = dev->data;
+	const struct nxp_enet_mac_config *config = dev->config;
+	enet_config_t enet_config;
+	uint32_t enet_module_clock_rate;
+#if defined(CONFIG_MDNS_RESPONDER) || defined(CONFIG_MDNS_RESOLVER)
+	/* standard multicast MAC address */
+	uint8_t mdns_multicast[6] = { 0x01, 0x00, 0x5E, 0x00, 0x00, 0xFB };
+#endif
+	int err;
+
+	err = pinctrl_apply_state(config->pincfg, PINCTRL_STATE_DEFAULT);
+	if (err) {
+		return err;
+	}
+
+	/* Initialize kernel objects */
+	k_mutex_init(&data->rx_frame_buf_mutex);
+	k_mutex_init(&data->tx_frame_buf_mutex);
+	k_sem_init(&data->rx_thread_sem, 0, CONFIG_ETH_NXP_ENET_RX_BUFFERS);
+	k_sem_init(&data->tx_buf_sem,
+		   CONFIG_ETH_NXP_ENET_TX_BUFFERS, CONFIG_ETH_NXP_ENET_TX_BUFFERS);
+#if defined(CONFIG_PTP_CLOCK_NXP_ENET)
+	k_sem_init(&data->ptp_ts_sem, 0, 1);
+#endif
+
+	if (config->generate_mac) {
+		config->generate_mac(data->mac_addr);
+	}
+
+	/* Start interruption-poll thread */
+	k_thread_create(&data->rx_thread, data->rx_thread_stack,
+			K_KERNEL_STACK_SIZEOF(data->rx_thread_stack),
+			eth_nxp_enet_rx_thread, (void *) dev, NULL, NULL,
+			K_PRIO_COOP(2),
+			0, K_NO_WAIT);
+	k_thread_name_set(&data->rx_thread, "eth_nxp_enet_rx");
+
+	/* Get ENET IP module clock rate */
+	err = clock_control_get_rate(config->clock_dev, config->clock_subsys,
+			&enet_module_clock_rate);
+	if (err) {
+		return err;
+	}
+
+	/* Use HAL to set up MAC configuration */
+	ENET_GetDefaultConfig(&enet_config);
+
+	if (IS_ENABLED(CONFIG_NET_PROMISCUOUS_MODE)) {
+		enet_config.macSpecialConfig |= kENET_ControlPromiscuousEnable;
+	}
+
+	if (IS_ENABLED(CONFIG_NET_VLAN)) {
+		enet_config.macSpecialConfig |= kENET_ControlVLANTagEnable;
+	}
+
+	if (IS_ENABLED(CONFIG_ETH_NXP_ENET_HW_ACCELERATION)) {
+		enet_config.txAccelerConfig |=
+			kENET_TxAccelIpCheckEnabled |
+			kENET_TxAccelProtoCheckEnabled;
+		enet_config.rxAccelerConfig |=
+			kENET_RxAccelIpCheckEnabled |
+			kENET_RxAccelProtoCheckEnabled;
+	}
+
+	enet_config.interrupt |= kENET_RxFrameInterrupt;
+	enet_config.interrupt |= kENET_TxFrameInterrupt;
+
+	if (config->phy_mode == NXP_ENET_MII_MODE) {
+		enet_config.miiMode = kENET_MiiMode;
+	} else if (config->phy_mode == NXP_ENET_RMII_MODE) {
+		enet_config.miiMode = kENET_RmiiMode;
+	} else {
+		return -EINVAL;
+	}
+
+	enet_config.callback = eth_callback;
+	enet_config.userData = (void *)dev;
+
+	ENET_Init(config->base,
+		  &data->enet_handle,
+		  &enet_config,
+		  &config->buffer_config,
+		  data->mac_addr,
+		  enet_module_clock_rate);
+
+#if defined(CONFIG_MDNS_RESPONDER) || defined(CONFIG_MDNS_RESOLVER)
+	ENET_AddMulticastGroup(config->base, mdns_multicast);
```

Comment:
This should no longer be necessary to add here, with the changes in `net_if_mcast_cb`.